File: 1304. Find N Unique Integers Sum up to Zero/Solution.py
class Solution:
    def sumZero(self, n: int) -> List[int]:
        ans = []
        # We will input pairs of positive and negative integer with same absolute value
        # so the total sum will become 0
        # loop will go till n//2 as there we will be adding 2 numbers in the array in one iteration
        for i in range(n//2):
            ans.append(i+1)
            ans.append(-(i+1))
        # if the number of integer required is odd then we will add a 0 in the array
        if n%2 != 0:
            ans.append(0)
        return ans

File: 1304. Find N Unique Integers Sum up to Zero/Solution.cpp
class Solution {
public:
    vector<int> sumZero(int n) {
        vector<int> ans;

        // Loop through half of n, since we add positive and negative pairs
        for(int i = 0; i < n / 2; i++)
        {
            ans.push_back(i + 1);   // add a positive number
            ans.push_back(-(i + 1)); // add its negative counterpart
        }

        // If n is odd, add 0 to make total count = n
        if(n % 2 != 0)
        {
            ans.push_back(0);
        }

        // Return the constructed vector
        return ans;
    }
};
